symengine/cwrapper.cpp

```diff
@@ -58,6 +58,25 @@ inline bool is_aligned(T *p, size_t n = alignof(T))
 
 extern "C" {
 
+
+#define CWRAPPER_BEGIN()                                                       \
+    try                                                                        \
+    {
+
+#define CWRAPPER_END(error, code)                                              \
+        return 0;                                                              \
+    }                                                                          \
+    catch ( error )                                                            \
+    {                                                                          \
+        return code;                                                           \
+    }
+
+#define CWRAPPER_END_EXTRA(error, code)                                        \
```

Comment:
I removed it and made a static CWRAPPER_END macro with all the current errors (actually `DivisionByZero` only). Will add more errors soon. Can you please check the new commit, especially the `symengine_exception.cpp` file?